From 5068b6f0082a020a6124cd7e652558efe3ec5f37 Mon Sep 17 00:00:00 2001
From: Sergey Edunov <edunov@fb.com>
Date: Fri, 10 Jun 2016 17:41:02 -0700
Subject: [PATCH] GIRAPH-1070 Comparators in PartitionUtils can overflow

Test Plan: mvn clean verify

Reviewers: majakabiljo, maja.kabiljo, dionysis.logothetis, heslami

Reviewed By: heslami

Differential Revision: https://reviews.facebook.net/D59547
---
 .../main/java/org/apache/giraph/partition/PartitionUtils.java | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/giraph-core/src/main/java/org/apache/giraph/partition/PartitionUtils.java b/giraph-core/src/main/java/org/apache/giraph/partition/PartitionUtils.java
index e4305ffe6..b607ed2be 100644
--- a/giraph-core/src/main/java/org/apache/giraph/partition/PartitionUtils.java
+++ b/giraph-core/src/main/java/org/apache/giraph/partition/PartitionUtils.java
@@ -64,7 +64,7 @@ private static class EdgeCountComparator implements
     @Override
     public int compare(Entry<WorkerInfo, VertexEdgeCount> worker1,
         Entry<WorkerInfo, VertexEdgeCount> worker2) {
-      return (int) (worker1.getValue().getEdgeCount() -
+      return Long.compare(worker1.getValue().getEdgeCount(),
         worker2.getValue().getEdgeCount());
     }
   }
@@ -81,7 +81,7 @@ private static class VertexCountComparator implements
     @Override
     public int compare(Entry<WorkerInfo, VertexEdgeCount> worker1,
         Entry<WorkerInfo, VertexEdgeCount> worker2) {
-      return (int) (worker1.getValue().getVertexCount() -
+      return Long.compare(worker1.getValue().getVertexCount(),
         worker2.getValue().getVertexCount());
     }
   }
